drivers/scheduler/k8s/k8s.go

```diff
@@ -5134,6 +5144,73 @@ func (k *K8s) createAdmissionRegistrationObjects(
 	return nil, nil
 }
 
+// createVirtualMachineObjects creates the kubevirt VirtualMachines using kubectl apply
+func (k *K8s) createVirtualMachineObjects(
+	spec interface{},
+	ns *corev1.Namespace,
+	app *spec.AppSpec,
+) (interface{}, error) {
+	if obj, ok := spec.(*kubevirtv1.VirtualMachine); ok {
+		// Validating to make sure the desired images are imported by CDI and volumes are ready to be used by Kubevirt VM
+		virtualMachineVolumes := obj.Spec.Template.Spec.Volumes
+		if len(virtualMachineVolumes) > 0 {
+			for _, v := range virtualMachineVolumes {
+				// Validating Volume Source of type PersistentVolumeClaim is ready to be used by Kubevirt VM
+				if v.VolumeSource.PersistentVolumeClaim != nil {
+					pvcName := v.VolumeSource.PersistentVolumeClaim.ClaimName
+					t := func() (interface{}, bool, error) {
+						events, err := k8sCore.ListEvents(ns.Name, metav1.ListOptions{
+							FieldSelector: fmt.Sprintf("involvedObject.kind=PersistentVolumeClaim,involvedObject.name=%s", pvcName),
+						})
+						if err != nil {
+							return "", false, err
+						}
+						log.Infof("Events for pvc [%s] in namespace [%s] for virtual machine [%s] \n%v\n", pvcName, ns.Name, obj.Name, events)
+						for _, event := range events.Items {
```

Comment:
I believe this block can be a separate method as well. If you believe this may be required by other future tests, then it would make sense to do so.

Reply:
We can but the function will need a lot of input parameters like FieldSelector, namespace, object name, key string etc.,
So I felt the function will become lengthy.